### PR TITLE
Perf improvements and fixes

### PR DESCRIPTION
**Pull request by edvald**
Caught some issues as I was working to accelerate the unit/functional test suite. The tests now run faster, startup time is improved, and some important issues were caught relating to version resolution.

Please review commit by commit.

### CODE REVIEW

File: core/src/logger/log-entry.ts [hunk 1]
```diff
@@ -32,7 +32,8 @@ export interface TaskMetadata {
   key: string
   status: TaskLogStatus
   uid: string
-  versionString: string
+  inputVersion: string
+  outputVersion?: string
```
Comment (Orzelius) [inline core/src/logger/log-entry.ts:36]: Okay, silly question, but what even are the task input and output versions?

**Review by Orzelius — APPROVED** [review]

File: core/src/vcs/git.ts [hunk 2]
```diff
@@ -379,7 +379,7 @@ export class GitHandler extends VcsHandler {
 
     await new Promise<void>((_resolve, _reject) => {
       const proc = lsFiles(this.ignoreFile)
-      proc.on("error", (err: execa.ExecaError) => {
+      void proc.on("error", (err: execa.ExecaError) => {
```
Comment (vvagaytsev) [inline core/src/vcs/git.ts:382]: I'm just curious, why do we need `void` here?

File: core/src/tasks/base.ts [hunk 3]
```diff
@@ -40,9 +40,7 @@ interface CommonTaskParams {
   skipDependencies?: boolean
 }
 
-export interface BaseTaskParams extends CommonTaskParams {
-  version: string
-}
+export interface BaseTaskParams extends CommonTaskParams {}
```
Comment (vvagaytsev) [inline core/src/tasks/base.ts:43]: This interface now looks redundant, shall we drop it?

**Review by vvagaytsev — APPROVED** [review]
LGTM! 🚀 I've left a few non-blocking comments.

File: core/src/logger/log-entry.ts (hunk 1 above)
Comment (edvald) [reply core/src/logger/log-entry.ts:36]: That's actually not at a silly question :) I'll explain this in comments, but basically an "input version" is generally the version of an action before it's resolved, the "output version" is e.g. after it's resolved, built etc. (which may in many cases transform the version).

File: core/test/unit/src/commands/custom.ts [hunk 4]
```diff
@@ -220,7 +220,7 @@ describe("CustomCommandWrapper", () => {
       variables: {
         foo: "test",
       },
-      gardenCommand: ["get", "doddi"],
+      gardenCommand: ["validate"],
```
[the remark this answers is not in the document]
Comment (edvald) [reply core/test/unit/src/commands/custom.ts:223]: Well, you can always run the command at your leisure .)

File: core/src/vcs/git.ts (hunk 2 above)
Comment (edvald) [reply core/src/vcs/git.ts:382]: I actually think the code was fine before, but the linter complained about a promise not being awaited.

**Pull request merged**